### PR TITLE
Change BFT to record all kv entries to the ledger

### PR DESCRIPTION
**Pull request by ashamis**
- There is an issue of with TPCC so it is being removed https://github.com/microsoft/CCF/issues/2662
- Do no record the shares table due to non-determinism https://github.com/microsoft/CCF/issues/2679

### CODE REVIEW

**Review by wintersteiger — APPROVED** [review]
Do I understand correctly that the main change here is that failed requests are now recorded where they weren't before?

File: src/consensus/aft/raft_tables.h [hunk 1]
```diff
@@ -9,14 +9,6 @@
...
-  static const std::unordered_set<std::string> replicated_tables_raft = {};
-
-  static constexpr auto replicate_type_bft = kv::ReplicateType::SOME;
-  static const std::unordered_set<std::string> replicated_tables_bft = {
-    ccf::Tables::AFT_REQUESTS,
-    ccf::Tables::SIGNATURES,
-    ccf::Tables::BACKUP_SIGNATURES,
-    ccf::Tables::NONCES,
-    ccf::Tables::NEW_VIEWS};
+  static constexpr auto replicate_type = kv::ReplicateType::ALL;
```
Comment (jumaffre) [inline src/consensus/aft/raft_tables.h:12]: Can we remove all the table replication filtering logic in the KV as well, and remove these types altogether? If we plan to use them again in the future, we should add a comment about that here.

File: src/kv/store.h [hunk 2]
```diff
@@ -842,6 +831,21 @@ namespace kv
             std::move(changes),
             std::move(new_maps));
         }
+        else if (
```
Comment (jumaffre) [inline src/kv/store.h:834]: Can we add a comment/link to https://github.com/microsoft/CCF/issues/2679 here?

**Review by achamayou — APPROVED** [review]

**Comment by ashamis** [comment]
> Do I understand correctly that the main change here is that failed requests are now recorded where they weren't before?

In this change a BFT network will record all writes to the key-value store. So in this change not committed requests will still not be recorded.

**Review by jumaffre — APPROVED** [review]

**Pull request merged**